include/questdb/ingress/line_sender.hpp

```diff
@@ -624,6 +626,39 @@ class line_sender_buffer
         return *this;
     }
 
+    /**
+     * Record a multidimensional double-precision array for the given column.
+     *
+     * @param name    Column name.
+     * @param shape   Array dimensions (e.g., [2,3] for a 2x3 matrix).
+     * @param data    Array data in row-major order. Size must match product of
+     * dimensions.
+     *
+     * @note Data is stored contiguously in row-major (C-style) order.
+     *       Example: shape [2,3] expects 6 elements ordered as:
+     *       [a11, a12, a13, a21, a22, a23]
+     */
+    template <typename T, size_t N>
+    line_sender_buffer& column(
+        column_name_view name,
+        const std::vector<uint32_t>& shape,
+        const std::array<T, N>& data)
```

Comment:
We need to come up with something that's a bit more idiomatic C++.
Let's park the C++ API for now until we have everything else nailed down.